### DISCUSSION TITLE
PickList issue with custom boolean column

### DISCUSSION THREAD

**Discussion by markomino** [primefaces/community#250] [PrimeFaces] [answered]
### Objective
I am trying to implement a picklist with custom functionality.
Custom button column **selectBooleanButton** which after selection performs custom behavior.

### Issue
After I **hover/select**  the button/checkbox all of the rows are affected. 

### Expected behavior
After hover/action only one row should be affected

### Questions

1. Do I need a custom converter for such column ?
2. Are columns in pickList supposed to be rather used as a single column ? 
(I am just guessing, since I've noticed in primefaces [example](https://www.primefaces.org/showcase/ui/data/pickList.xhtml?jfwid=45378) that country icon is added as one column, when I tried to use it as second column the styling gets messy.

### My code:

    
    <p:pickList id="countryPickList" value="#{bean.countrySelector}" var="country"
                widgetVar="countryPickList"
                itemValue="#{country}" itemLabel="#{country.name}"
                responsive="true"
                converter="pickListConverter">

        <f:facet name="sourceCaption">#{admin['adm.country.available']}</f:facet>
        <f:facet name="targetCaption">#{admin['adm.country.current']} </f:facet>

        <p:column>
            <h:outputText value="#{country.name}"/>
        </p:column>

        <p:column>
            <p:graphicImage name="#{country.imageUrl}"/>
        </p:column>

        <p:column>
                <p:selectBooleanButton value="#{country.main}"/>
        </p:column>

    </p:pickList>

**Comment by melloware** [comment, marked as answer]
Picklist is not meant to have input fields like SelectBooleanButton in ts columns. Its not only bad UI/UX but its not how the component was designed.
